src/chainstate/stacks/db/blocks.rs

```diff
@@ -4245,6 +4253,30 @@ impl StacksChainState {
                 )
             };
 
+            // get previous burn block stats
+            let (parent_burn_block_hash, parent_burn_block_height, parent_burn_block_timestamp) =
+                if block.is_first_mined() {
+                    (BurnchainHeaderHash([0; 32]), 0, 0)
+                } else {
+                    match SortitionDB::get_block_snapshot_consensus(
+                        burn_dbconn,
+                        &parent_consensus_hash,
+                    )? {
+                        Some(sn) => (
+                            sn.burn_header_hash,
+                            sn.block_height as u32,
+                            sn.burn_header_timestamp,
+                        ),
+                        None => {
+                            // shouldn't happen
+                            panic!(
```

Comment:
This panic code is reachable under normal conditions.  If we're doing a PoX reorg, then because this isn't carried out in a transaction, it's possible that the coordinator thread will invalidate the block snapshot just before `get_block_snapshot_consensus()` gets called, leading to the `None` case.

Reply:
Generally speaking, it's a Bad Idea to panic on network code paths.

Reply:
Ok, will remove the panic, and use an option type for the burn block stats instead. 

Also, I have questions about the invalidation logic you mention. The function `get_block_snapshot_consensus` queries with `SELECT * FROM snapshots WHERE consensus_hash = ?1`, and the invalidation function I saw just updates the field `pox_valid` to 0. Also, this code block is (I'm pretty sure), being called in the coordinator thread. I'm unsure in what cases that `get_block_snapshot_consensus` would return `None` here.

Reply:
@jcnelson

Reply:
Ah, I stand corrected.  I had forgotten that this method ignores the `pox_valid` flag.

But just to be safe, let's remove the `panic!()` and log a warning.  I've had a couple sleepless nights now that can basically be summarized as stories in three acts:

Act 1: "There's no way this code is reachable"
Act 2: "The life, death, and resurrection of the blockchain"
Act 3: "It was totally the unreachable code"